app/main.rs

```diff
@@ -852,29 +862,67 @@ async fn main() -> Result<()> {
         Either::Left(validator)
     };
 
-    let (task, err_rxs) = task(enforcer.clone(), cli, mainchain_client, info.chain).await;
+    let (original_task_handle, mut err_rxs) =
+        task(enforcer.clone(), cli, mainchain_client, info.chain).await;
+
+    // TODO: make this a shared future. ErrReport is not cloneable?
+    let shared_task_handle = original_task_handle.map(|res| {
+        // The JoinHandle itself resolves to Result<InnerResult, JoinError>
+        // We need to flatten this for the shared future.
+        match res {
+            Ok(Ok(())) => Ok(()),
+            Ok(Err(err)) => Err(err),
+            Err(join_error) => Err(miette!(
+                "main task panicked or was cancelled: {join_error:#}"
+            )),
+        }
+    });
```

Comment:
This is currently non-shared future. I want to make this cloneable, so I can pass it into `handles` on line 881, as well as selecting on it on line 918. Unable to get the compiler to let me do so...